api/src/test/java/org/openmrs/module/fgh/mpi/MpiContextTest.java

```diff
@@ -0,0 +1,127 @@
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		mpiContext.setAuthenticationType(AUTHENTICATION_TYPE);
+	}
+	
+	@Test
+	public void context_shouldInitOauth() throws Exception {
+		AuthenticationType OUAUTH = AuthenticationType.OAUTH;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(OUAUTH.toString());
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		MpiContext initOauthContext = new MpiContext();
+		initOauthContext.init();
+		
+		assertNotNull(initOauthContext.getAuthenticationType());
+		assertEquals(OUAUTH, initOauthContext.getAuthenticationType());
+		assertEquals(MPI_BASE_URL, initOauthContext.getServerBaseUrl());
+		assertEquals(MPI_SYSTEM, initOauthContext.getMpiSystem());
+		assertEquals(UUID_SYSTEM, initOauthContext.getOpenmrsUuidSystem());
+		assertEquals(SANTE_CLIENT_ID, initOauthContext.getClientId());
+		assertEquals(SANTE_CLIENT_SECRET, initOauthContext.getClientSecret());
+		assertTrue(initOauthContext.isContextInitialized());
+	}
+	
+	@Test
+	public void context_shouldInitSSL() throws Exception {
+		AuthenticationType CERTIFICATE = AuthenticationType.CERTIFICATE;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(CERTIFICATE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM_AS_OPENCR.toString());
+		when(adminService.getGlobalProperty(GP_KEYSTORE_PATH)).thenReturn("src/test/resources/log4j.xml");
+		when(adminService.getGlobalProperty(GP_KEYSTORE_PASS)).thenReturn(GP_KEYSTORE_PASS);
+		when(adminService.getGlobalProperty(GP_KEYSTORE_TYPE)).thenReturn(GP_KEYSTORE_TYPE);
+		when(KeyStore.getInstance(any())).thenReturn(null);
+		when(KeyManagerFactory.getInstance("SunX509")).thenReturn(null);
+		MpiContext initSSLContext = new MpiContext();
+		
+		try {
+			initSSLContext.init();
+		}
+		catch (Exception e) {
```

Comment:
Why do you need the catch block? I think you should remove it, it an exception is thrown, then the test must fail